File: src/frontend/models/Animal.js
/**
 * Animal Model
 */
class Animal {
  /**
   * @param {string} name - Name of the animal
   * @param {string} type - Type of animal (dog, cat, bird, etc.)
   * @param {string|null} breed - Breed of the animal
   * @param {number|null} age - Age in years
   */
  constructor(name, type, breed = null, age = null) {
    this.name = name;
    this.type = type;
    this.breed = breed;
    this.age = age;
  }

  /**
   * Create Animal from JSON object
   * @param {Object} json 
   * @returns {Animal}
   */
  static fromJSON(json) {
    return new Animal(
      json.name,
      json.type,
      json.breed || null,
      json.age || null
    );
  }

  /**
   * Convert to plain object
   * @returns {Object}
   */
  toJSON() {
    return {
      name: this.name,
      type: this.type,
      breed: this.breed,
      age: this.age
    };
  }
}

module.exports = Animal;

File: src/agent/models/animal.py
from pydantic import BaseModel, Field
from typing import Optional
from enum import Enum


class AnimalType(str, Enum):
    DOG = "dog"
    CAT = "cat"
    BIRD = "bird"
    RABBIT = "rabbit"
    HAMSTER = "hamster"
    OTHER = "other"


class Animal(BaseModel):
    name: str = Field(..., description="Name of the animal")
    type: AnimalType = Field(..., description="Type of animal")
    breed: Optional[str] = Field(None, description="Breed of the animal")
    age: Optional[int] = Field(None, ge=0, description="Age in years")